checkAuth.js

```diff
@@ -0,0 +1,39 @@
+import jwt from "jsonwebtoken";
+import axios from "axios";
+import jwkToPem from "jwk-to-pem";
+
+const generatePolicy = (principalId, effect, resource) => {
+  const authResponse = {};
+  authResponse.principalId = principalId;
+  if (effect && resource) {
+    const policyDocument = {};
+    policyDocument.Version = "2012-10-17";
+    policyDocument.Statement = [];
+    const statementOne = {};
+    statementOne.Action = "execute-api:Invoke";
+    statementOne.Effect = effect;
+    statementOne.Resource = resource;
+    policyDocument.Statement[0] = statementOne;
+    authResponse.policyDocument = policyDocument;
+  }
+  return authResponse;
+};
+export const auth = async (event, context) => {
+  const token = event.authorizationToken;
+
+  if (!token) return "Not authorized, no token!!";
+  const tokenParts = event.authorizationToken.split(" ");
+  const tokenValue = tokenParts[1];
+
+  let result = await axios.get(
+    `https://cognito-idp.${process.env.AWS_REGION}.amazonaws.com/${process.env.AWS_USER_POOL_ID}/.well-known/jwks.json`
+  );
+  const pem = jwkToPem(result.data.keys[1]);
+  try {
+    const decoded = jwt.verify(tokenValue, pem, { algorithm: ["RS256"] });
```

Comment:
what are we actually verifying here, that the token came from our cognito?

question: should anyone be able to edit a hack? or only the hacks original author? or a team mate?

Reply:
Yes we are verifying the token pass by Cognito which will be attached to our axios request.

At the moment we haven't 100% decided who can edit the hack, for the time being only the login user, I think Joseph have some logic in the front end to determine that if the hack belongs to a certain user. But then again after this token verification might have to further implement logic to guard against unauthorised edit